src/client/datascience/interactive-window/nativeInteractiveWindow.ts

```diff
@@ -121,7 +128,8 @@ export class NativeInteractiveWindow implements IInteractiveWindowLoadable {
         private _notebookUri: Uri | undefined, // This remains the same for the lifetime of the InteractiveWindow object
```

Comment:
Can we refactor this contstructor.
Today `_notebookUri` is undefined & it only gets set inside the contructor after the interactive window is opened via the command `interactive.open`.
I'd prefer we did that out side the contructor, this way 

as follows:
```typescript
const notebookUri = await this.commandManager
            .executeCommand('interactive.open', ViewColumn.Beside)
            .then((result) => result.notebookUri);


const xyx = new NativeInteractiveWindow(...., notebookUri).

```

This way, the property `NativeInteractiveWindow.notebookUri` is never undefined.
Else it gets too complicated to figure out & handle situations where this can be undefined & not... 

I'd prefer if we could just create the interactive window outside this class (in the `nativeInteractiveWindowProvider.create` method).
Makes it more cleaner & precise.

Reply:
+1 on this. The lifecyle of the `NativeInteractiveWindow` should probably be tied to `NotebookDocument` and then the Uri is never undefined.

Reply:
Maybe introduce a static `NativeInteractiveWindow.create()` if we want to embed the logic of iw opening inside `NativeInteractiveWindow`